### PR TITLE
fix: vehicle link

### PR DESCRIPTION
**Pull request by Marine-Berthier**
[IN-1271](https://autoricardo.atlassian.net/browse/IN-1271?atlOrigin=eyJpIjoiYzllNjk1MjY5OTg5NDhjZjhkYTdlNGY0MmQxYmYyMWIiLCJwIjoiaiJ9)

## Motivation and context

Chad noticed that the link "Meine Fahrzeug" takes us to  prod even though we are on int

## Before

Screenshot 2023-09-07 at 15 45 29

## After

I added a condition to return the right link but I am not sure we want to do this. 
Please let me know what you think. 

[IN-1271]: https://autoricardo.atlassian.net/browse/IN-1271?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by lkappeler — COMMENTED** [review]
Maybe we can generalize the approach a bit more as there are more cases and we might miss them in the future.

Screenshot 2023-09-08 at 13 44 04
Screenshot 2023-09-08 at 13 43 33

Current setup
image
image

What I am thinking is to extend the domain config (naming and structure is just a rough idea to showcase what I have in mind..)

```
 this.domains = {
        autoscout24: {
          main: {
            production: 'www.autoscout24.ch',
            preprod: 'int.autoscout24.ch',
          }
          internal: [
            {
              production: 'dealer.autoscout24.ch',
              preprod: 'int-dealer.autoscout24.ch',
            },
            {
              production: 'my.autoscout24.ch',
              preprod: 'my-dealer.autoscout24.ch',
            }
          ]
        },
        motoscout24: {
          main: {
            production: 'www.motoscout24.ch',
            preprod: 'int.motoscout24.ch',
          }
          internal: [
            {
              production: 'dealer.autoscout24.ch',
              preprod: 'int-dealer.autoscout24.ch',
            },
            {
              production: 'my.autoscout24.ch',
              preprod: 'my-dealer.autoscout24.ch',
            }
          ]
        },
      },
    };
```

And then extend the base link to replace all instances of internal production domains
```

    const updatedLink = this.prefixDomain({
      link: config.link,
      brand,
      environment,
      useAbsoluteUrls,
      linkProtocol,
      domains,
    });
    this.link = replaceInternlaDomains(updatedLink)

  }

  private replaceInternlaDomains(...) {
    // if env === production return
    // if current link starts with production domain replace with evn domain 
  }
  ```

**Review by lkappeler — APPROVED** [review]
Nice job 👏 I like the solution you came up with 👍 

No need to change it! An observation on a conceptual level to provide you with some insights.
The concept you went for is quite interlocked with the subdomains we have currently as internal links are tied to a specific user type be aware that this will make it harder to extend in the future should the requirements for that come up. 
As an example, we have `int-autoradar.autoscout24.ch` where the main will be neither of the ones we have so far. When extending it for that case we would need to add additional branches and eventually props to the links so we can determine which case the current link represents.
When for example picking the replacement approach we will not face that problem and can simply extend the domains/case to as many as we want with very few changes needed to the code we have in place. It's always a hard balance between how much abstraction makes sense in terms of additional logic required, how likely it is to change and how much simpler and faster a very concrete implementation is

**Comment by Milos-Zbiljic** [comment]
> What I am thinking is to extend the domain config (naming and structure is just a rough idea to showcase what I have in mind..)

This approach could limit as in future (but maybe that is ok), as we are limited to only `dealer` | `my` subdomains. I am not sure if we should think a bit more about solution or go with the current one.

File: src/components/navigation/header/config/headerNavigationLink.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export class HeaderNavigationLink extends Link {
   variant?: 'navigationLink' | 'subNavigationLink';
   color?: string;
   userAvatar?: ReactNode;
+  isInternal?: boolean;
```
[the remark this answers is not in the document]
Comment (Marine-Berthier) [reply src/components/navigation/header/config/headerNavigationLink.ts]: Yes we don't!

**Review by Milos-Zbiljic — APPROVED** [review]

**Pull request merged**